Helper/PaymentResponseHandler.php

```diff
@@ -356,6 +360,56 @@ public function handlePaymentsDetailsResponse(
         return $result;
     }
 
+    /**
+     * @param $order
+     * @return void
+     * @throws NoSuchEntityException
+     */
+    public function cancelOrder($order): void
```

Comment:
This method handles a business logic related to an order. Even though, it's part of the response handling logic, this operation is not this class' responsibility. I believe it can be moved to `Order` helper.